labgrid/driver/power/gude24.py

```diff
@@ -1,49 +1,73 @@
+
+Driver has been tested with:
+* Gude Expert Power Control 8080
+'''
+
 import requests
 
 from ..exception import ExecutionError
 
 
-# This driver implements a power port for Gude Power Switches with up to
-# 24 ports.
-# These switches differ in their API to the previous 8-port switches for set-
-# and get-commands.
-#
-# Driver has been tested with:
-# * Gude Expert Power Control 8080
-
 PORT = 80
 
 def power_set(host, port, index, value):
-    # The gude web-interface uses different pages for the three groups of
-    # switches. The web-interface always uses the 'correct' page to set a
-    # value. But commands for all pages are accepted on all pages.
+    '''
+    The gude web-interface uses different pages for the three groups of
+    switches. The web-interface always uses the 'correct' page to set a
+    value. But commands for all pages are accepted on all pages.
+    '''
     index = int(index)
     assert 1 <= index <= 24
     # access the web interface...
     value = 1 if value else 0
-    r = requests.get(
+    response = requests.get(
         "http://{}:{}/ov.html?cmd=1&p={}&s={}".format(host, port, index, value)
     )
-    r.raise_for_status()
+
+    # Check, that the port is in the desired state
+    state = get_state(response, index)
+    if state != value:
+        raise ExecutionError("failed to set port {} to status {}".format(index, value))
 
 
 def power_get(host, port, index):
-    # The status of the ports is made available via a html <meta>-tag using the
-    # following format:
-    # <meta http-equiv="powerstate" content="Power Port 1,0">
-    # Again the status of all ports is made available on all pages.
+    '''
+    Get the status of a port.
+    '''
+    # get the contents of the main page
+    response = requests.get("http://{}:{}/ov.html".format(host, port))
+    state = get_state(response, index)
+    return state
+
+
+def get_state(request, index):
+    '''
+    The status of the ports is made available via a html <meta>-tag using the
+    following format:
+    <meta http-equiv="powerstate" content="Power Port 1,0">
+    Again the status of all ports is made available on all pages.
+    '''
+    request.raise_for_status()
+
+    # Get the power state of a specified index
+    # raise an exception if the state cannot be determined.
     index = int(index)
     assert 1 <= index <= 24
-    # get the contents of the main page
-    r = requests.get("http://{}:{}/ov.html".format(host, port))
-    r.raise_for_status()
-    for line in r.text.splitlines():
+
+    for line in request.text.splitlines():
         if line.find("content=\"Power Port {}".format(index)) > 0:
             if line.find(",0") > 0:
                 return False
             if line.find(",1") > 0:
                 return True
 
             raise ExecutionError("failed to parse the port status")
+
     # if we got this far, something is wrong with the website
-    raise ExecutionError("failed to find the port")
+    print(request.text)
```

Comment:
leftover debug print.

Reply:
Still it might be good to get the response of the request to see why it failed. In our case, request.text contains the reason (device is locked) as well as the ip of the user that locked the device.